src/Webapi/Dom/Webapi__Dom__HtmlMediaElement.re

```diff
@@ -0,0 +1,139 @@
+    | NetworkNoSource; // 3
+
+  let decodeNetworkState = fun
+    | 0 => Some(NetworkEmpty)
+    | 1 => Some(NetworkIdle)
+    | 2 => Some(NetworkLoading)
+    | 3 => Some(NetworkNoSource)
+    | _ => None;
+
+  [@bs.get] external _networkState: t_htmlMediaElement => int = "networkState";
+  let networkState = t_htmlMediaElement => _networkState(t_htmlMediaElement) |> decodeNetworkState;
+
+  [@bs.get] external paused: t_htmlMediaElement => bool = "paused";
+
+  [@bs.get] external playbackRate: t_htmlMediaElement => float = "playbackRate";
+  [@bs.set] external setPlaybackRate: (t_htmlMediaElement, float) => unit = "playbackRate";
+
+  [@bs.get] external played: t_htmlMediaElement => Webapi__TimeRanges.t = "played";
+  [@bs.get] external preload: t_htmlMediaElement => string = "preload";
+
+  type readyStateEnum =
+    | HaveNothing
+    | HaveMetadata
+    | HaveCurrentData
+    | HaveFutureData
+    | HaveEnoughData;
+
+  let decodeReadyState = fun
+    | 0 => Some(HaveNothing)
+    | 1 => Some(HaveMetadata)
+    | 2 => Some(HaveCurrentData)
+    | 3 => Some(HaveFutureData)
+    | 4 => Some(HaveEnoughData)
+    | _ => None;
+
+  [@bs.get] external _readyState: t_htmlMediaElement => int = "readyState";
+  let readyState = t => _readyState(t) |> decodeReadyState;
+
+  [@bs.get] external seekable: t_htmlMediaElement => Webapi__TimeRanges.t = "seekable";
+  [@bs.get] external seeking: t_htmlMediaElement => bool = "seeking";
+  // [@bs.get] external sinkId: t_htmlMediaElement => string = "sinkId"; // experimental
+
+  [@bs.get] external src: t_htmlMediaElement => string = "src";
+  [@bs.set] external setSrc: (t_htmlMediaElement, string) => unit = "src";
+
+  [@bs.get] [@bs.return nullable] external srcObject: t_htmlMediaElement => option(mediaProvider) = "srcObject";
+
+  [@bs.get] external audioTracks: t_htmlMediaElement => Webapi__AudioTrackList.t = "textTracks";
+  [@bs.get] external textTracks: t_htmlMediaElement => Webapi__TextTrackList.t = "textTracks";
+
+  // [@bs.get] external videoTracks: t_htmlMediaElement => Webapi__VideoTrackList.t = "videoTracks"; // Not widely available
+
+  [@bs.get] external volume: t_htmlMediaElement => float = "volume";
+  [@bs.set] external setVolume: (t_htmlMediaElement, float) => unit = "volume";
+
+  /** Methods */
+
+  [@bs.send] external addTextTrack: (t_htmlMediaElement, Webapi__TextTrack.textTrackKind, ~label: string, ~language: string, unit) => Webapi__TextTrack.t = "addTextTrack";
+
+  // return is one of "probably", "maybe", or ""
+  [@bs.send] external canPlayType: (t_htmlMediaElement, string) => string = "canPlayType";
+  // [@bs.send] external fastSeek: (t_htmlMediaElement, float) => unit = "fastSeek"; // experimental
+  [@bs.send] external load: t_htmlMediaElement => unit = "load";
+  [@bs.send] external pause: t_htmlMediaElement => unit = "pause";
+  [@bs.send] external play: t_htmlMediaElement => Js.promise(unit, string /* [`NotAllowedError | `NotSupportedError] */) = "play";
```

Comment:
This is the deprecated promise type. The rest of this library is using `Js.Promise.t`

Reply:
Thanks!